RosNet/RosMessageParser/MessageDefinitionParser.cs

```diff
@@ -157,7 +157,7 @@ private List<FieldValue> ParseDefinition(IEnumerable<string> lines, Dictionary<s
                         }
                         else
                         {
-                            throw new KeyNotFoundException($"Could not find {arrayDataTypeString}");
+                            throw new KeyNotFoundException($"The dataType of array: { arrayDataTypeString } is not a primitive type or defined in MessageDefinition");
```

Comment:
```suggestion
                            throw new KeyNotFoundException($"The dataType of array: {arrayDataTypeString} is not a primitive type or defined in MessageDefinition");
```

Reply:
Also the change of the happy and bad path here, so that the _exception_ is handled first, then keeping the normal course of action as the left-most indented was just closed without handling it? I might just be a bit too early with the review though